uaclient/util.py

```diff
@@ -352,11 +352,18 @@ def get_platform_info() -> "Dict[str, str]":
 @lru_cache(maxsize=None)
 def is_container(run_path: str = "/run") -> bool:
     """Checks to see if this code running in a container of some sort"""
+    try:
+        subp(["ischroot"])
+        return False
+    except ProcessExecutionError:
```

Comment:
Curious as to why here we're catching a `ProcessExecutionError` where down below we're catching `IOError` and `OSError` - do you know what is causing the difference?

Reply:
The `ProcessExecutionError` happens when the command return something different from the expected return codes, which by default is `0`. I don't have the context of the `systemd` call here, but I assume that the command fails for different reasons then the expected return code

Reply:
Got it - thanks!